src/wasm2js.h

```diff
@@ -2446,6 +2541,67 @@ void Wasm2JSGlue::emitSpecialSupport() {
   function wasm2js_data_drop(segment) {
     // TODO: traps on invalid things
     memorySegments[segment] = new Uint8Array(0);
+  }
+      )";
+    } else if (import->base == ABI::wasm2js::ATOMIC_WAIT_I32) {
+      out << R"(
+  function wasm2js_atomic_wait_i32(ptr, expected, timeoutLow, timeoutHigh) {
+    if (timeoutLow != -1 || timeoutHigh != -1) throw 'unsupported timeout';
+    var view = new Int32Array(bufferView.buffer); // TODO cache
+    var result = Atomics.wait(view, ptr, expected);
+    if (result == 'ok') return 0;
+    if (result == 'not-equal') return 1;
+    if (result == 'timed-out') return 2;
+    throw 'bad result ' + result;
+  }
+      )";
+    } else if (import->base == ABI::wasm2js::ATOMIC_RMW_I64) {
+      out << R"(
+  function wasm2js_atomic_rmw_i64(op, bytes, offset, ptr, valueLow, valueHigh) {
+    assert(bytes == 8); // TODO
+    var view = new BigInt64Array(bufferView.buffer); // TODO cache
+    ptr = (ptr + offset) >> 3;
+    var value = BigInt(valueLow >>> 0) | (BigInt(valueHigh >>> 0) << BigInt(32));
```

Comment:
I guess this assumes the browser has BigInt but no BigInt WebAssembly integration? Does that describe most browsers?

Reply:
I think so, yeah, https://caniuse.com/#feat=bigint vs https://webassembly.org/roadmap/

But why do you say this code assumes there is no bigint wasm integration?

Reply:
Oh, because it doesn't return a bigint? Yeah, the code assumes wasm MVP.